### ISSUE TITLE
Disappearing cursor

### ISSUE THREAD

**Issue by dnkl** [swaywm/sway#3617]
wlroots: https://github.com/swaywm/wlroots/commit/20d404a091fa92f56be4825bbaf89e4027fda154
sway: ec5da0ca5bad6a433f727499d68ac1352397f5aa

Not sure if this is a problem with sway, or not. But, I'm now using sway on multiple systems, with the same software setup.

Everything is working fine on all systems except one, where the mouse cursor keeps disappearing. It seems its position is correctly tracked though, because e.g. hover effects in browsers work, focus-follows-mouse works and so, on. It's just that the cursor itself is invisible.

The cursor is usually invisible right from the start, before any windows have been opened. Then, if I move it between windows, it will sometimes become visible. When that happens, it stays visible until moved to another window.

The sway debug log doesn't contain any obvious error messages, except these:
```
2019-02-07 21:22:15 - [sway/input/cursor.c:1132] denying request to set cursor from unfocused client
```
and
```
glamor: No eglstream capable devices found
```

The main difference between the non-working system and the working ones are that the non-working one has an nvidia card (with nouveau), while the others are intel based.

**Comment by RedSoxFan** [comment]
Related https://github.com/swaywm/wlroots/pull/1526

**Comment by I60R** [comment]
I have the same issue: `Quadro 3000m + nouveau`, `Linux 4.20, Manjaro`.

**Edit:** setting `WLR_NO_HARDWARE_CURSORS=1` env. variable fixes it

**Comment by dnkl** [comment]
This is probably established fact by now, but just wanted to let you know that https://github.com/swaywm/wlroots/pull/1537 (which I found through https://github.com/swaywm/wlroots/pull/1526) works for me too.

**Comment by LamaRaz** [comment]
I have an Ivy Bridge CPU/gpu and I also had this problem, with cursor disappearing.

WLR_NO_HARDWARE_CURSORS fixed it but surely this bug ought to have a better fix?

**Comment by tarkov2213** [comment]
WLR_NO_HARDWARE_CURSORS doesn't help me

I have a peculiar problem where the cursor disappears sometimes (usually after closing mpv but sometimes also randomly) and it can still be used just not seen. But there is always a **way to make it come back by guiding it to the status bar** (i3status-rust) and it will always pop back to visibility.

Is there a solution?

Also, at precisely 5 seconds of no activity the cursor disappears, but it always comes back when moving the mouse. Is that a feature and how to turn it off?

**Comment by mamutal91** [comment]
Coming back to say that the problem happened again

**Comment by cshapeshifter** [comment]
I'm also having this problem, but on intel graphics and `WLR_NO_HARDWARE_CURSORS=1` does not fix it.
For example, the cursor would suddenly become invisible over firefox, but become visible when moved over an edge or to another window. I noticed that changing the scale factor will sometimes bring back to the cursor over one application, but it would still become invisible over some other application. Normally, I use scale 3, and sometimes, the cursor would be tiny (as in scale 1) instead of being invisible.

**Comment by nboughton** [comment]
I'm having this issue on my laptop (radeon graphics) but it only occurs after I unlock from swaylock and only if the mouse pointer was on my external monitor when I locked the desktop. 

Then it will only appear on the external monitor. Sway is definitely aware of where the cursor is as I can see the focus change on the laptop monitor when I move the mouse around but the cursor is invisible.

**Comment by emersion** [comment]
Can someone try wlroots 0.14.1 and post new debug logs?

**Comment by toto-dev** [comment]
@emersion I have a debug log but it contains some personal information do you have some script to anonymize it a bit, like removing window titiles?

**Comment by emersion** [comment]
Nope. Can you at least share the `[ERROR]` lines to confirm this is the bug I think this is?

**Comment by toto-dev** [comment]
Here you have, just the error lines: [sway.error.log](https://github.com/swaywm/sway/files/6865041/sway.error.log)

**Comment by emersion** [comment]
Alright, that's what I've been hitting as well:

```
41237:02:14:10.874 [ERROR] [wlr] [render/swapchain.c:97] No free output buffer slot
41238:02:14:10.874 [ERROR] [wlr] [types/wlr_output.c:1342] Failed to render cursor buffer
```

**Comment by emanuelserpa** [comment]
Yeah, I'm having this same problem on Haswell.

![image](https://user-images.githubusercontent.com/934124/126880746-1a5d7a6b-e189-4703-8244-621de0eaa418.png)

**Comment by ChazyTheBest** [comment]
Same here, I'm on arch linux. I'm using 2 monitors one using display port the other hdmi. The cursor disappears only in the display port monitor, when i move the cursor to the other one i can see the cursor just fine.

**Comment by CatalinPuscoci** [comment]
I'm running into this issue as well with a HDMI monitor. And I'm having the opposite problem, the cursors is invisible only in the main display, being visible on the external one. going into a terminal and opening wdisplays seems to fix it

**Comment by charims** [comment]
Unplugging and replugging in the monitor fixed it for me when I unlocked my desktop system this morning and the right monitor cursor was invisible. It was DP-1

```
$ lspci | grep VGA
03:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT] (rev ca)
```

```
$ swaymsg -t get_outputs
Output HDMI-A-1 'Dell Inc. DELL U2414H FN8GP68S1JML'
  Current mode: 1920x1080 @ 60.000 Hz
  Position: 0,0
  Scale factor: 1.000000
  Scale filter: nearest
  Subpixel hinting: unknown
  Transform: normal
  Workspace: 5
  Max render time: off
  Adaptive sync: disabled
  Available modes:
    720x400 @ 70.082 Hz
    640x480 @ 59.940 Hz
    640x480 @ 59.940 Hz
    640x480 @ 60.000 Hz
    640x480 @ 75.000 Hz
    720x480 @ 59.940 Hz
    720x480 @ 59.940 Hz
    720x480 @ 59.940 Hz
    720x480 @ 60.000 Hz
    720x480 @ 60.000 Hz
    720x576 @ 50.000 Hz
    800x600 @ 60.317 Hz
    800x600 @ 75.000 Hz
    1024x768 @ 60.004 Hz
    1024x768 @ 75.029 Hz
    1280x720 @ 50.000 Hz
    1280x720 @ 59.940 Hz
    1280x720 @ 60.000 Hz
    1280x720 @ 60.000 Hz
    1152x864 @ 75.000 Hz
    1280x800 @ 60.000 Hz
    1440x900 @ 60.000 Hz
    1280x1024 @ 60.020 Hz
    1280x1024 @ 75.025 Hz
    1600x900 @ 60.000 Hz
    1680x1050 @ 60.000 Hz
    1600x1200 @ 60.000 Hz
    1920x1080 @ 50.000 Hz
    1920x1080 @ 59.940 Hz
    1920x1080 @ 60.000 Hz
    1920x1080 @ 60.000 Hz

Output DP-1 'Dell Inc. DELL U2414H FN8GP68S1HEL' (focused)
  Current mode: 1920x1080 @ 60.000 Hz
  Position: 1920,0
  Scale factor: 1.000000
  Scale filter: nearest
  Subpixel hinting: unknown
  Transform: normal
  Workspace: 2
  Max render time: off
  Adaptive sync: disabled
  Available modes:
    720x400 @ 70.082 Hz
    640x480 @ 59.940 Hz
    640x480 @ 59.940 Hz
    640x480 @ 60.000 Hz
    640x480 @ 75.000 Hz
    720x480 @ 59.940 Hz
    720x480 @ 59.940 Hz
    720x480 @ 59.940 Hz
    720x480 @ 60.000 Hz
    720x480 @ 60.000 Hz
    720x576 @ 50.000 Hz
    800x600 @ 60.317 Hz
    800x600 @ 75.000 Hz
    1024x768 @ 60.004 Hz
    1024x768 @ 75.029 Hz
    1280x720 @ 50.000 Hz
    1280x720 @ 59.940 Hz
    1280x720 @ 60.000 Hz
    1280x720 @ 60.000 Hz
    1152x864 @ 75.000 Hz
    1280x800 @ 60.000 Hz
    1440x900 @ 60.000 Hz
    1280x1024 @ 60.020 Hz
    1280x1024 @ 75.025 Hz
    1600x900 @ 60.000 Hz
    1680x1050 @ 60.000 Hz
    1600x1200 @ 60.000 Hz
    1920x1080 @ 50.000 Hz
    1920x1080 @ 59.940 Hz
    1920x1080 @ 60.000 Hz
    1920x1080 @ 60.000 Hz
```

**Comment by sykhro** [comment]
Had the same thing happen this morning on a DP monitor (dual-monitor setup).

**Comment by HeptaSean** [comment]
Have the same issue consistently after `swayidle` takes both the laptop monitor LVDS-1 and the external monitor HDMI-A-1 off and turns them on again. Mouse cursor is missing on LVDS-1, but is present on HDMI-A-1, afterwards. As observed by others, effects of the mouse are there (mouse overs are highlighted, clicks are possible), but there is just no pointer/cursor.

Messages by sway are lots of:
>01:15:54.682 [ERROR] [wlr] [render/swapchain.c:97] No free output buffer slot
>01:15:54.682 [ERROR] [wlr] [types/wlr_output.c:1342] Failed to render cursor buffer

**Comment by emanuelserpa** [comment]
Is this a regression? I don't remember having this problem on 1.5.x.
